src/coreclr/interpreter/intrinsics.cpp

```diff
@@ -0,0 +1,137 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "interpreter.h"
+#include "intrinsics.h"
+
+#define HAS_PREFIX(haystack, needle) \
+    (strncmp(haystack, needle, strlen(needle)) == 0)
+
+NamedIntrinsic GetNamedIntrinsic(COMP_HANDLE compHnd, CORINFO_METHOD_HANDLE compMethod, CORINFO_METHOD_HANDLE method, bool enableFallback)
+{
+    const char* className = NULL;
+    const char* namespaceName = NULL;
+    const char* methodName = compHnd->getMethodNameFromMetadata(method, &className, &namespaceName, NULL, 0);
+
+    // Array methods don't have metadata
+    if (!namespaceName)
+        return NI_Illegal;
+
+    if (!HAS_PREFIX(namespaceName, "System"))
+        return NI_Illegal;
+
+    if (!strcmp(namespaceName, "System"))
+    {
+        if (!strcmp(className, "Double") || !strcmp(className, "Single"))
```

Comment:
```suggestion
        if (enableFallback && (!strcmp(className, "Double") || !strcmp(className, "Single")))
```

Reply:
After thinking about it some more, we may want to disable all intrinsic handling in the interpreter when the fallback is enabled so that we produce the exact same results as the JITed code.

Reply:
I'll change it to work that way, good idea.